apps/wolfsshd/configuration.c

```diff
@@ -503,49 +505,80 @@ static int HandleInclude(WOLFSSHD_CONFIG *conf, const char *value)
             prefixLen = (int)(ptr - value);
         }
 
-        d = opendir(path);
-        if (d) {
-            while ((dir = readdir(d)) != NULL) {
-                if (dir->d_type == DT_DIR) {
-                    /* Skip sub-directories */
-                    continue;
+        if (!WOPENDIR(NULL, conf->heap, &d, path)) {
+            word32 fileCount = 0, i, j;
+            char** fileNames = NULL;
+
+            /* Count up the number of files */
+            while ((dir = WREADDIR(&d)) != NULL) {
+                /* Skip sub-directories */
+                if (dir->d_type != DT_DIR) {
+                    fileCount++;
                 }
-                else {
-                    /* Check if filename prefix matches */
-                    if (prefixLen > 0) {
-                        if ((int)WSTRLEN(dir->d_name) <= prefixLen) {
-                            continue;
-                        }
-                        if (WSTRNCMP(dir->d_name, prefix, prefixLen) != 0) {
-                            continue;
+            }
+            WREWINDDIR(&d);
+
+            if (fileCount > 0) {
+                fileNames = (char**)WMALLOC(fileCount * sizeof(char*), NULL, 0);
```

Comment:
Can use `conf->heap` for the heap argument to `WMALLOC` here and elsewhere.

Reply:
And added dynamic types.